packages/editor/src/browser/editor-widget.ts

```diff
@@ -8,14 +8,17 @@
 import { SelectionService } from '@theia/core/lib/common';
 import { Widget, BaseWidget, Message } from '@theia/core/lib/browser';
 import { TextEditor } from "./editor";
+import { FileIconProvider } from '@theia/filesystem/lib/browser/icons/file-icons';
 
 export class EditorWidget extends BaseWidget {
 
     constructor(
         readonly editor: TextEditor,
-        readonly selectionService: SelectionService
+        readonly selectionService: SelectionService,
+        readonly iconProvider: FileIconProvider
     ) {
         super(editor);
+        this.title.iconClass = iconProvider.getFileIconForURI(this.editor.uri);
```

Comment:
maybe it should be done outside: https://github.com/theia-ide/theia/pull/573/files#diff-4fde0d93433e07c55d6c127ee3e86d76R116

it is just some title's attributes assigned outside, other here, looks inconsistent.